Sources/Commands/SwiftTool.swift

```diff
@@ -472,22 +472,34 @@ public class SwiftTool {
     }
 
     private func getCachePath(fileSystem: FileSystem = localFileSystem) throws -> AbsolutePath? {
-        // Create the default cache directory.
-        let idiomaticCachePath = fileSystem.swiftPMCacheDirectory
-        if !fileSystem.exists(idiomaticCachePath) {
-            try fileSystem.createDirectory(idiomaticCachePath, recursive: true)
-        }
-        // Create ~/.swiftpm if necessary
-        if !fileSystem.exists(fileSystem.dotSwiftPM) {
-            try fileSystem.createDirectory(fileSystem.dotSwiftPM, recursive: true)
+        if options.skipCache {
+            return nil
         }
-        // Create ~/.swiftpm/cache symlink if necessary
-        let dotSwiftPMCachesPath = fileSystem.dotSwiftPM.appending(component: "cache")
-        if !fileSystem.exists(dotSwiftPMCachesPath, followSymlink: false) {
-            try fileSystem.createSymbolicLink(dotSwiftPMCachesPath, pointingAt: idiomaticCachePath, relative: false)
+
+        if let explicitCachePath = options.cachePath {
+            return explicitCachePath
```

Comment:
discussed with @abertelrud and @neonichu decided to try to create and error out of fails